service/history/workflow/update/update_test.go

```diff
@@ -336,6 +337,82 @@ func TestRequestSendReject(t *testing.T) {
 	})
 }
 
+func TestAdmitAcceptOrReject(t *testing.T) {
+	// See comment in onAcceptanceMsg about stateAdmitted.
+	// This test is to ensure that transition from stateAdmitted to stateAccepted or stateCompleted (rejected) is valid.
+
+	t.Parallel()
+	var (
+		ctx     = context.Background()
+		effects = effect.Immediate(ctx)
+		meta    = updatepb.Meta{UpdateId: t.Name() + "-update-id"}
+		req     = updatepb.Request{Meta: &meta, Input: &updatepb.Input{Name: t.Name()}}
+		acpt    = protocolpb.Message{Body: mustMarshalAny(t, &updatepb.Acceptance{
+			AcceptedRequestMessageId:         "random",
+			AcceptedRequestSequencingEventId: 2208,
+		})}
+		acceptedEventData = struct {
+			updateID                         string
+			acceptedRequestMessageId         string
+			acceptedRequestSequencingEventId int64
+			acceptedRequest                  *updatepb.Request
+		}{}
```

Comment:
It seems this isn't actually used anywhere (the writes in the store don't seem to have an effect)?